### PR TITLE
Adding in rudimentary benchmark system

### PR DESCRIPTION
**Pull request by Pauan**
As per issue #59, this is the first step in improving the performance of the queue implementation.

### CODE REVIEW

**Comment by kriskowal** [comment]
I am tempted to just land this because I’m so glad you’re on board, but I’m going to provide style feedback anyway so I can communicate how you can save me some time in the future.

File: benchmarks/benchmarks.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+var drain = require("./drain");
+var benchmark = require("./scaffold");
+
+benchmark.run();
```
Comment (kriskowal) [inline benchmarks/benchmarks.js]: If we call this file `index.js`, we can run `node benchmarks` to execute them all.

File: benchmarks/scaffold.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+var Benchmark = require("benchmark");
+
+var suite = new Benchmark.Suite({
+    onCycle: function (event) {
+        console.log("" + event.target);
+    }
+});
+
+function add_timer(s, f) {
```
Comment (kriskowal) [inline benchmarks/scaffold.js]: dromedaryCase.

File: benchmarks/scaffold.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    suite.add(s, function (deferred) {
+        f(function () {
```
Comment (kriskowal) [inline benchmarks/scaffold.js]: I should add a rule to the linter for named functions.

File: package.json [hunk 4]
```diff
@@ -49,6 +49,7 @@
     "q-io": "^2.0.3",
     "saucelabs": "^0.1.1",
     "wd": "^0.2.21",
-    "weak-map": "^1.0.5"
+    "weak-map": "^1.0.5",
+    "benchmark": "^1.0.0"
```
Comment (kriskowal) [inline package.json]: You can add a `"scripts.benchmarks": "node benchmarks"` so we can do `npm run benchmarks`

**Comment by Pauan** [comment]
Okay, I fixed the issues you brought up.

**Comment by kriskowal** [comment]
Looks like CI broke. I’ll have to swing back and fix that.

**Pull request merged**